crates/anvil/src/eth/api.rs

```diff
@@ -1825,6 +1830,36 @@ impl EthApi {
         Ok(())
     }
 
+    /// Deals ERC20 tokens to a address
+    ///
+    /// Handler for RPC call: `anvil_dealERC20`
+    pub async fn anvil_deal_erc20(
+        &self,
+        address: Address,
+        balance: U256,
+        token_address: Address,
+    ) -> Result<()> {
+        node_info!("anvil_dealERC20");
+
+        sol! {
+            #[sol(rpc)]
+            contract IERC20 {
+                function balanceOf(address target) external view returns (uint256);
+            }
+        }
+
+        let calldata = IERC20::balanceOfCall { target: address }.abi_encode();
+        let tx = TransactionRequest::default().with_input(calldata);
+        let access_list_result = self.create_access_list(WithOtherFields::new(tx), None).await?;
+        let access_list = access_list_result.access_list;
+        for item in access_list.0 {
+            if item.address != token_address {
+                continue;
+            }
+        }
```

Comment:
now here we need to loop over all slots and send an ethcall with a stateoverride for the slot in order to find the slot that has the balance:

https://github.com/Rubilmax/viem-deal/blob/cf378cedcb28bbcc18fcbfac2c6f9e5c333edcf3/src/actions/test/deal.ts#L106-L108

and then finally we need to call set storage for that slot and value